codec-multipart/src/main/java/io/netty/contrib/handler/codec/http/multipart/HttpPostRequestEncoder.java

```diff
@@ -789,14 +785,11 @@ public HttpRequest finalizeRequest() throws ErrorDataEncoderException {
         headers.set(HttpHeaderNames.CONTENT_LENGTH, String.valueOf(realSize));
         if (realSize > HttpPostBodyUtil.chunkSize || isMultipart) {
             isChunked = true;
-            if (transferEncoding != null) {
-                headers.remove(HttpHeaderNames.TRANSFER_ENCODING);
-                for (CharSequence v : transferEncoding) {
-                    if (HttpHeaderValues.CHUNKED.contentEqualsIgnoreCase(v)) {
-                        // ignore
-                    } else {
-                        headers.add(HttpHeaderNames.TRANSFER_ENCODING, v);
-                    }
+            while (transferEncoding.hasNext()) {
+                CharSequence v = transferEncoding.next();
```

Comment:
I'm not sure the transfer-encoding parts get broken out into separate header values anymore.
It might be better to query with `boolean contains(CharSequence, CharSequence)` which specifically handles comma-separated values.

Reply:
Thanks for pointing out this.
Now, while reviewing the problematic code you have mentioned, I see that what was done is the following:

in _finalizeRequest_, for any Transfer-Encoding entries, the old code is first trying to manually exclude any existing chunked TE like [this](https://github.com/netty-contrib/codec-multipart/blob/main/codec-multipart/src/main/java/io/netty/contrib/handler/codec/http/multipart/HttpPostRequestEncoder.java#L792-L801), so here the old code first removes all Transfer-Encoding headers, then any existing TE entries which are not chunked are re-added ([see here](https://github.com/netty-contrib/codec-multipart/blob/main/codec-multipart/src/main/java/io/netty/contrib/handler/codec/http/multipart/HttpPostRequestEncoder.java#L798)).

But, then , after that, the old code is calling the _HttpUtil.setTransferEncodingChunked_ method which **overrides** all existing Transfer-Encoding entries with one single Transfer-Encoding: chunked entry, see [here](https://github.com/netty-contrib/codec-multipart/blob/main/codec-multipart/src/main/java/io/netty/contrib/handler/codec/http/multipart/HttpPostRequestEncoder.java#L802), and [here](https://github.com/netty/netty/blob/4.1/codec-http/src/main/java/io/netty/handler/codec/http/HttpUtil.java#L334)

So, in the last commit, I have simply removed the old code which was extracting any chunked entries before calling the _HttpUtil.setTransferEncodingChunked_, this is not needed, since the HttpUtil.setTransferEncodingChunked overrides any existing Transfer-Encoding header values.

is it ok, or am I missing something else ?